### ISSUE TITLE
[Feature Request] Object Previews in Editor

### ISSUE THREAD

**Issue by AdamDN** [SmartlyDressedGames/Unturned-3.x-Community#3345]
I'm sure this has been suggested in the past, but something like this would be a massive time saver for map editors... 

https://user-images.githubusercontent.com/71527679/181923416-67c0d984-42e0-4b53-96b5-05a55c97ef34.mp4

**Comment by GazziFX** [comment]
wtf is that, why you didnt tried it directly in Unturned?

**Comment by AdamDN** [comment]
> wtf is that, why you didnt tried it directly in Unturned?
Unity..? And because it's much easier to throw together a quick mock-up on top of a screenshot of the editor, rather than describing the features/mechanics/positions/sizes etc in a lengthy post. Seeing it in action is much more effective and getting the suggestion across.

**Comment by GazziFX** [comment]
I mean you can easy implement it in game

**Comment by surv0013** [comment]
Good idea, but I'm afraid the rendering of the model may be slow, it would be better to just show image. I don’t think that Nelson will do this in the near future, he has been writing for quite some time that he wants to rework UI.

**Comment by AdamDN** [comment]
> I'm afraid the rendering of the model may be slow, it would be better to just show image.

Spawning/deleting the model in the render window shouldn't cause any noticeable slowdown unless the model is extremely detailed, and even then it should be no different (if not more efficient) than manually spawning/deleting the object into the map to check it's appearance. And generating an image would still involve rendering the model afaik.

**Comment by surv0013** [comment]
> Spawning/deleting the model in the render window shouldn't cause any noticeable slowdown unless the model is extremely detailed, and even then it should be no different (if not more efficient) than manually spawning/deleting the object into the map to check it's appearance. And generating an image would still involve rendering the model afaik.

He wants the window to appear on hover, but what if I quickly hover over all the list items?

Of course, the image will include the rendering of the model, but you can create all the images in advance, even at the stage of entering the editor, and simply save them in the game's memory.

_This topic can be discussed for a long time, but the system load directly depends on the implementation of this idea. I see no point in discussing something that will not happen in the near future._

**Comment by DanielWillett** [comment]
Then he could make it togglable in settings, or make it show up after you've been hovering for half a second. Rendering a model like that, especially at a lower resolution, doesn't take that long. An entire inventory's worth of icons only takes a few hundred ms to render.

**Comment by SDGNelson** [comment]
Closing because the issues list was not really intended for feature requests. Sorry to close this discussion - I do WANT to fulfill all (most?) requests, but it is a matter of prioritization. Constantly being reminded of all the pending requests when checking issues is stressful. I will keep the request in mind. On GitHub maybe the Ideas section of Discussions would be the best place.
